tests/src/core/testqgsogcutils.cpp

```diff
@@ -464,6 +467,56 @@ void TestQgsOgcUtils::testExpressionFromOgcFilter()
   QCOMPARE( dumpText, expr->expression() );
 }
 
+void TestQgsOgcUtils::testExpressionFromOgcFilterWithLongLong_data()
+{
+  QTest::addColumn<QString>( "xmlText" );
+  QTest::addColumn<QString>( "dumpText" );
+  QTest::newRow( "Literal less than" ) << QString(
+                                         "<Filter><And>"
+                                         "<PropertyIsGreaterThan>"
+                                         "<PropertyName>id</PropertyName>"
+                                         "<Literal>1</Literal>"
+                                         "</PropertyIsGreaterThan>"
+                                         "<PropertyIsLessThan>"
+                                         "<PropertyName>id</PropertyName>"
+                                         "<Literal>3</Literal>"
+                                         "</PropertyIsLessThan>"
+                                         "</And></Filter>" )
+                                       << QStringLiteral( "id > 1 AND id < 3" );
+}
+
+void TestQgsOgcUtils::testExpressionFromOgcFilterWithLongLong()
+{
+  QFETCH( QString, xmlText );
+  QFETCH( QString, dumpText );
+
+  QDomDocument doc;
+
+  QVERIFY( doc.setContent( xmlText, true ) );
+  QDomElement rootElem = doc.documentElement();
+
+  QgsVectorLayer layer( "Point?crs=epsg:4326", "temp", "memory" );
+
+  QgsField longlongField( QStringLiteral( "id" ), QVariant::LongLong );
+
+  QList<QgsField> fields;
+  fields.append( longlongField );
+  layer.dataProvider()->addAttributes( fields );
+  layer.updateFields();
+
+  std::shared_ptr<QgsExpression> expr( QgsOgcUtils::expressionFromOgcFilter( rootElem, &layer ) );
```

Comment:
Totally irrelevant because it's inside a test, but still: is there a good reason to use a shared pointer for a local variable?

Reply:
>  is there a good reason to use a shared pointer for a local variable?

None. I've just copy/paste the test function just above :).

Reply:
can you make it a `unique_ptr` :)

Reply:
Of course, done